packages/front-end/components/Metrics/MetricForm/TypeaheadInput.tsx

```diff
@@ -0,0 +1,97 @@
+import { useRef, useState } from "react";
+import CreatableSelect from "react-select/creatable";
+
+type Props = {
+  currentValue: string;
+  onChange: (label: string, value?: string) => void;
+  placeholder?: string;
+  label?: string;
+  options: { label: string; value: string }[];
+};
+
+export default function TypeaheadInput({
+  currentValue,
+  onChange,
+  label,
+  placeholder,
+  options,
+}: Props) {
+  const [inputValue, setInputValue] = useState("");
+
+  const inputRef = useRef(null);
+
+  function currentOption(): { label: string; value: string } | undefined {
+    if (!currentValue) return undefined;
+
+    return (
+      options.find((item) => item.label === currentValue) || {
+        label: currentValue,
+        value: "",
+      }
+    );
+  }
+
+  return (
+    <>
```

Comment:
I would have this wrap in a div with a `form-group` class. Also, this component should probably also accept some more of the common fields used by SelectField like helpText.

Reply:
@jdorn I added `helpText` and `required`, let me know if there are any else you think we need.

Re: The `form-group` class - this is added one-level up within the Metric Form's `index` file - adding it here caused some weird spacing, specifically with the `conditions` group.